### PR TITLE
fix(ci): Optimize CI performance

### PR DESCRIPTION
**Pull request by ahmed-nour-fdc**
This PR includes the following:
  * Do not save built images locally when running in the CI.
  * Cancel stale commits' runs when a new commit is pushed.
  * Install docker with apt in integration tests instead of earthly installing it every time.

### CODE REVIEW

File: .github/workflows/execution-plan-pr.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ name: "Execution Plan on Pull Request"
   pull_request:
     branches:
       - main
+concurrency: 
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```
Comment (sven-urbanski-freiheit-com) [inline .github/workflows/execution-plan-pr.yml:8]: huh, I thought that is already how it worked. I guess I was wrong.

File: services/cd-service/Makefile [hunk 2]
```diff
@@ -64,11 +64,11 @@ build: bin/main
 
 build-pr:
 	echo "build on pull request"
-	$(EARTHLY) --push +build-pr --registry=$(IMAGE_REGISTRY) --tag=$(VERSION) --mirror=$(ARTIFACT_REGISTRY_MIRROR)
+	$(EARTHLY) --no-output --push +build-pr --registry=$(IMAGE_REGISTRY) --tag=$(VERSION) --mirror=$(ARTIFACT_REGISTRY_MIRROR)
```
Comment (sven-urbanski-freiheit-com) [inline services/cd-service/Makefile:67]: does this mean that the image is not pushed to the registry anymore?
That would disrupt @miguel-crespo-fdc's current workflow.

File: .github/workflows/execution-plan-pr.yml (hunk 1 above)
Comment (ahmed-nour-fdc) [reply .github/workflows/execution-plan-pr.yml:8]: No it never cancelled runs for stale commits.

File: services/cd-service/Makefile (hunk 2 above)
Comment (ahmed-nour-fdc) [reply services/cd-service/Makefile:67]: It gets pushed, but it does not get saved locally. Check these [logs](https://github.com/freiheit-com/kuberpult/actions/runs/8754669556/job/24027084007#step:12:358)
Images are pushed, but not saved locally.

**Review by sven-urbanski-freiheit-com — APPROVED** [review]

**Pull request merged**